sdk/mobile-center-crashes/src/main/java/com/microsoft/azure/mobile/crashes/Crashes.java

```diff
@@ -182,6 +197,7 @@ private Crashes() {
         thread.start();
         mHandler = new Handler(thread.getLooper());
         mLastSessionCrashProcessingStatus = PREPARE_PROCESSING;
+        mSemaphore = new Semaphore(0);
```

Comment:
This is wrong to use a shared semaphore if you have more than 1 caller. This is not the right synchronization mechanism to use in that case. Use a Condition or a CountDownLatch (I prefer CountDownLatch as it does not have finally block, finally is troublesome with code coverage).